src/auto-clickers/auto-clicker.js

```diff
@@ -1,42 +1,128 @@
+import { autoclickerManifest } from "./auto-clicker-manifest.js";
+import { CONSTANTS, OBJECT_TYPES } from "../constants/constants.js";
+
+export async function getAutoclickerClass(className, scene) {
+    let path = autoclickerManifest[className].classPath;
+    let clickerClass = await import(path);
+
+    return new clickerClass.default(scene);
+}
+
 export class AutoClicker {
+    objectType = OBJECT_TYPES.AUTOCLICKER;
     name = "";
+    examineText = "";
+    cost = 0;
     dps = 0;
     level = 0;
 
-    scene;
+    // Scenes
+    scrollWindow;
+    chat;
+    dashboard;
+    currentScene;
+    stats;
 
     damageInterval = 0;
     timer = {};
+    text = {};
 
     constructor(data) {
-        console.log("Creating " + data.name + " autoclicker with " + data.dps + " dps");
-        // Pull into local objects
-        this.scene = data.scene;
-        this.stats = data.scene.stats;
+        this.scrollWindow = data.scene;
+        this.examineText = data.examineText;
+        this.cost = data.cost;
         this.dps = data.dps;
         this.level = data.level;
-        this.name = name.type;
+        this.name = data.name;
 
-        // Damage every .1 second
-        this.damageInterval = 0.1;
-        this.timer = setInterval(this.clickTarget.bind(this), this.damageInterval * 1000);
+        // Damage every .5 seconds
```

Comment:
Isn't it still .1 seconds?

Reply:
Lol yeah. I think I liked .1s better so i reverted it but didn't update the comment